From dcb04ae498c16f90b3a4376ad5674156c3ec0dc6 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 7 Feb 2021 23:58:41 -0800
Subject: [PATCH] llvm: Do not use find_library for ncurses

This ensures that it lets OE to decide which lib to link
otherwise it adds absolute paths to linker cmdline and confuses it
horribly with native and target libs when build clang for target

TOPDIR/build/tmp/work/cortexa57-yoe-linux-musl/clang/12.0.0-r0/recipe-sysroot-native/usr/lib/libtinfo.so: error adding symbols: file in wrong format
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 compiler-rt/cmake/config-ix.cmake | 2 +-
 llvm/cmake/config-ix.cmake        | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/compiler-rt/cmake/config-ix.cmake b/compiler-rt/cmake/config-ix.cmake
index 196aa62fd01c..49bbbd257d97 100644
--- a/compiler-rt/cmake/config-ix.cmake
+++ b/compiler-rt/cmake/config-ix.cmake
@@ -140,7 +140,7 @@ else()
   set(MAYBE_REQUIRED)
 endif()
 if(LLVM_ENABLE_TERMINFO)
-  find_library(COMPILER_RT_TERMINFO_LIB NAMES terminfo tinfo curses ncurses ncursesw ${MAYBE_REQUIRED})
+  set(TERMINFO_LIB tinfo curses ncurses ncursesw)
 endif()
 if(COMPILER_RT_TERMINFO_LIB)
   set(LLVM_ENABLE_TERMINFO 1)
diff --git a/llvm/cmake/config-ix.cmake b/llvm/cmake/config-ix.cmake
index 818fafbce148..2f8ad6652334 100644
--- a/llvm/cmake/config-ix.cmake
+++ b/llvm/cmake/config-ix.cmake
@@ -175,7 +175,7 @@ if(NOT LLVM_USE_SANITIZER MATCHES "Memory.*")
       set(MAYBE_REQUIRED)
     endif()
     if(LLVM_ENABLE_TERMINFO)
-      find_library(TERMINFO_LIB NAMES terminfo tinfo curses ncurses ncursesw ${MAYBE_REQUIRED})
+      set(TERMINFO_LIB tinfo curses ncurses ncursesw)
     endif()
     if(TERMINFO_LIB)
       set(LLVM_ENABLE_TERMINFO 1)
